File: OOP/Car/Python/Car.py
from abc import ABC, abstractmethod

class Vehicle(ABC):
    @abstractmethod
    def start(self): ...

class Car(Vehicle):
    def __init__(self, engine_id):
        self.engine_id = engine_id
    def start(self):
        print(f"Car {self.engine_id} starts!")

v = Car("eng-001")
v.start()        

File: OOP/Car/Go/Car.go
package main

import "fmt"

type Vehicle interface {
	Start()
}

type Car struct {
	engineID string
}

func (c Car) Start() {
	fmt.Printf("Car %s starts!\n", c.engineID)
}

func main() {
	var v Vehicle = Car{engineID: "V8"}
	v.Start()
}
